### ISSUE TITLE
Project maintenance

### ISSUE THREAD

**Issue by nvsnvikram** [ymnk/jzlib#22]
We are planning to use JZlib in our project. I wanted to know the project status as there is not much activity on issues or code updates. Can you let me know the status of this project?

**Comment by sunny-shu** [comment]
@ymnk 
Is this project still under maintenance，If needed, I can help you maintain this project，I am familiar with zlib

**Comment by headius** [comment]
We are also willing to help maintain this library. In particular, we need to add JPMS module info to support JRuby modularity going forward (https://github.com/jruby/jruby/issues/4835).

Given the lack of recent activity, it may be necessary to officially fork jzlib to a new organization.

**Comment by headius** [comment]
I have sent email to info@jcraft to try to reach @ymnk and find out the plans for this library. If I hear back I will reply here.

**Comment by headius** [comment]
I never received a response to my email so I'm going to proceed to take over maintenance and duplicate this library on my account. I'll post here when I have it up and working.

**Comment by norrisjeremy** [comment]
Hi @headius,

FYI, we have embedded a version of JZlib into our JSch fork over at https://github.com/mwiede/jsch/tree/master/src/main/java/com/jcraft/jsch/jzlib.

We tweaked it a bit (reformatted code and changed visibility of several classes & methods) since our intent is to only use it for SSH compression purposes within JSch and not provide a general purpose Zlib implementation for downstream consumers.

One major item we also performed recently was to rewrite the unit tests from Scala to Java (see https://github.com/mwiede/jsch/tree/master/src/test/java/com/jcraft/jsch/jzlib).

Not sure if any of the changes we made could be of interest to your fork.

Thanks,
Jeremy

**Comment by norrisjeremy** [comment]
HI @headius,

Also, I'm not sure if you noticed that the dev branch of this repo (https://github.com/ymnk/jzlib/tree/dev) has a few additional commits in it (not sure if they would be applicable to your fork or not)?

Thanks,
Jeremy

**Comment by headius** [comment]
Yes, in fact I rediscovered this issue while looking for a fix to the modification time bug. There's also a pull request on that branch that claims to support concatenated GZip files, but as it broke some ruby tests I have reverted it on my branch to revisit later.

I welcome contributors! https://github.com/jruby/jzlib

**Comment by He-Pin** [comment]
@headius Does it possible to rewrite it in pure scala, then scala-native can use it too.

**Comment by headius** [comment]
@He-Pin An interesting idea, but I don't think porting the entire library to Scala is a very high priority. Perhaps there are some Java to Scala converters that might do the trick?

**Comment by He-Pin** [comment]
@headius yes, the problems is it's hard to build a zlib on Windows.

A converter will work, especially with bow days Ai, but it should be better to be maintained somewhere with love.

**Comment by headius** [comment]
@He-Pin Perhaps you should try to convert it and start a separate repository? You could maintain it with love.
